cli/init.go

```diff
@@ -0,0 +1,451 @@
+package cli
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	githubTarballURL = "https://github.com/%s/tarball/main"
+	githubAPIURL     = "https://api.github.com/repos/%s/branches/main"
+	repo             = "dispatchrun/dispatch-templates"
+	dispatchUserDir  = "dispatch"
+)
+
+func directoryExists(path string) (bool, error) {
+	info, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	return info.IsDir(), nil
+}
+
+func isDirectoryEmpty(path string) (bool, error) {
+	dir, err := os.Open(path)
+	if err != nil {
+		return false, err
+	}
+	defer dir.Close()
+
+	// Read directory names, limiting to one to check if it's not empty
+	_, err = dir.Readdirnames(1)
+	if err == nil {
+		// The directory is not empty
+		return false, nil
+	}
+	if err == io.EOF {
+		// The directory is empty
+		return true, nil
+	}
+	// Some other error occurred
+	return true, err
```

Comment:
I would suggest returning `false` here since that's the zero-value, it's a more natural default in the condition where we don't know whether the directory is empty or not.
```suggestion
	return false, err
```

Reply:
fixed